### PR TITLE
RF: do not raise if git is not configured -- just issue a warning

### LINKED ISSUE #2272: DataLad requires git configuration
#### What is the problem?
I'm creating a docker image with a dataset inside (using `pip` to install `datalad[full]`. It was working last week and  today I'm getting an error "[ERROR] You must configure git first (set both user.name and user.email) before using DataLad. [gitrepo.py:check_git_configured:349] (RuntimeError) "
I was wondering if this is an intended change.

#### What steps will reproduce the problem?

This is my [Dockerfile](https://github.com/miykael/nipype_tutorial/blob/master/Dockerfile) and here is [failing CircleCi](https://circleci.com/gh/miykael/nipype_tutorial/11)

This is [Circle CI report from last week](https://circleci.com/gh/djarecka/nipype_tutorial/47?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

### PR DESCRIPTION
**Pull request by yarikoptic**
This pull request fixes #2272 

This pull request proposes to not fail if there is no git configuration -- just issue a warning.

The rationale is that git became even more "robust" in assigning some author name/email, and annex does some tricks too...  but if we always demand configuration -  that just complicates the simplest consumer usage.

### Changes
- [x] This change is complete

Please have a look @datalad/developers

### CODE REVIEW

File: datalad/support/gitrepo.py [hunk 1]
```diff
@@ -338,17 +338,20 @@ def check_git_configured():
...
-                "You must configure git first (set both user.name and "
-                "user.email) before using DataLad."
-            )
+            exc_ += exc_str(exc)
+    if exc_:
+        lgr.warning(
+            "It is highly recommended to configure git first (set both "
+            "user.name and user.email) before using DataLad. Failed to "
+            "verify that git is configured: %s.  Some operations might fail or "
+            "not perform correctly."
```
Comment (kyleam) [inline datalad/support/gitrepo.py]: Did you forget to pass `exc_` as an argument to the format string?

Comment (yarikoptic) [reply datalad/support/gitrepo.py]: sure I did, thank you @kyleam , will amend!

**Comment by kyleam** [comment]
Peeking at the python 3.4 (DATALAD_REPO_VERSION=5) log, there are four
failing tests:

```
FAIL: datalad.distribution.tests.test_uninstall.test_uninstall_recursive
FAIL: datalad.distribution.tests.test_uninstall.test_kill
FAIL: datalad.interface.tests.test_add_archive_content.test_add_archive_content
FAIL: datalad.support.tests.test_gitrepo.test_check_git_configured
```

The last one is obviously related:

```
======================================================================
FAIL: datalad.support.tests.test_gitrepo.test_check_git_configured
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/datalad/datalad/datalad/tests/utils.py", line 627, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "/home/travis/build/datalad/datalad/datalad/support/tests/test_gitrepo.py", line 1228, in test_check_git_configured
    assert_raises(RuntimeError, GitRepo, newhome, create=True)
AssertionError: RuntimeError not raised by GitRepo
```

I've deleted this test in 0b3ffd306.

**Comment by yarikoptic** [comment]
I have added a few CPed commits from master which resolved failing unrelated tests due to newer git-annex being used.

**Comment by yarikoptic** [comment]
All green is a rare occasion, so merging since no objections and we should kick out a quick bug fix release

**Pull request merged**